src/http/client.cr

```diff
@@ -295,6 +296,16 @@ class HTTP::Client
     self.read_timeout = read_timeout.total_seconds
   end
 
+  # Sets the number of seconds to wait for one block to be written (via one `write(2)` call) before raising an `IO::Timeout`.
```

Comment:
I would prefer a description that does not involve syscall and more from the users perspective, like `read_timeout=` one.

Out of curiosity, other than the client suddenly not using the established connection, in which scenarios this is needed?

Reply:
I copied the description from Ruby, though I'm open for improvement suggestions :)

Reply:
It bugs me that write_timeout can be understood for any of the directions. Is the client not writing or the server not writing. The same applies to the read_timeout. But the sample code there shows no content written. Here that is not evident since the sample is only in the spec.

I'm pointing as a consumer of the documentation in mind, without knowledge of syscall.

My (optional) suggestion would be to clarify in which situation the timeout could raise and why. And to remove the mention to the syscall.

Reply:
> Sets the write timeout in seconds - the maximum time you allow to write (send) the request data to the service before raising an `IO::Timeout`.

Would that be better?

Reply:
@bcardiff @RX14 ping

Reply:
It's the timeout while waiting to make *any progress* sending the request, not for sending the whole request.

Reply:
>Sets the write timeout in seconds - if any chunk of response is not written within number of seconds provided to write_timeout, `IO::Timeout` exception is raised.

how about now?

Reply:
(Sorry the delay, I’m on vacations) looks better, but either I am miss understanding something or I would expect to be: any chunk of the request, but not the response.

Isn’t the spec asserting that the client is not able to send the request body in the required amount of time?

Reply:
I think that's correct, I've updated my previous comment.

Reply:
@RX14 @bcardiff ping

Reply:
I still prefer "if no progress is made sending the request data for the number of seconds provided to write_timeout"

Reply:
I'm bit lost, could you please edit the docs in this PR to your liking and, since it's approved possibly merge it already? Would be a shame to keep monkey-patching that just because it was too hard to sync on the content of the docs - for over 2 week - which can always be improved later on...

Reply:
Just put whats in your comment in, I'm not too bothered.

Reply:
My intention is to have this method understandably documented, so if you have a full comment text I could put there for everyone to be happy with it I'd most happy to do that.

Reply:
@RX14 @bcardiff ready for re-review.